### PR TITLE
feat: add option to exclude dir(s) from compiling to assests catalog …

### PR DESCRIPTION
**Pull request by mbender74**
…when app-thinning is used

needs:  https://github.com/tidev/node-titanium-sdk/pull/581

tiapp.xml:
```
 <ios>
        <exclude-dir-from-asset-catalog>tiles</exclude-dir-from-asset-catalog>
```

### CODE REVIEW

File: cli/lib/gather.js [hunk 1]
```diff
@@ -297,14 +300,23 @@ class Categorizer {
 					// if we are using app thinning, then don't copy the image, instead mark the
 					// image to be injected into the asset catalog. Also, exclude images that are
 					// managed by their bundles.
+
 					if (this.useAppThinning && !relPath.match(BUNDLE_FILE_REGEXP)) {
-						results.imageAssets.set(relPath, info);
-						return;
+						if (this.excludeAssestsDir !== null) {
+							const checkRegEx = new RegExp(this.excludeAssestsDir);
```
Comment (hansemannn) [inline cli/lib/gather.js:306]: Can you fix the regex warning?

Comment (mbender74) [reply cli/lib/gather.js:306]: sure... on it...

Comment (mbender74) [reply cli/lib/gather.js]: no regex warning....
also added missing part for "png"
Usage in tiapp.xml (no "/" at the start!!!):

```
<ios>
        <exclude-dir-from-asset-catalog>maps/tiles|maps/otherimages</exclude-dir-from-asset-catalog>
```

**Comment by mbender74** [comment]
@hansemannn yeah.... now I can PR my ti.map stuff.... ;-)

**Comment by hansemannn** [comment]
@mbender74 This PR causes our app to crash with latest master. I think I found the root cause and will setup a PR now.

File: cli/lib/gather.js [hunk 2]
```diff
@@ -268,6 +271,17 @@ class Categorizer {
...
+						let testPath = this.excludeAssestsDir;
+						const checkRegEx = new RegExp(`${testPath}`);
+						if (!relPath.match(checkRegEx)) {
+							results.imageAssets.set(relPath, info);
+							return;
+						}
+					} else {
+						results.imageAssets.set(relPath, info);
+						return;
+					}
```
Comment (hansemannn) [inline cli/lib/gather.js:284]: This else-case did not exist before, so it should be removed.

**Pull request merged**